### PR TITLE
Basic quickchick tactic for use during proofs

### PR DESCRIPTION
**Pull request by segevem**
Adds a tactic for Quickchecking the current proof goal in a naive fashion, using default generators for all variables and checking and filtering inputs against the hypotheses.

### CODE REVIEW

**Comment by liyishuai** [comment]
The CI is failing. Which version of Coq are you working on?
If backwards compatibility requires excessive effort, maybe worth dropping support for some old versions of Coq.

**Comment by Lysxia** [comment]
This looks impressive!

**Comment by segevem** [comment]
> The CI is failing. Which version of Coq are you working on? If backwards compatibility requires excessive effort, maybe worth dropping support for some old versions of Coq.

The CI was failing due to some warnings I had to resolve, I'm using Coq 8.17.0

**Comment by segevem** [comment]
Discovered why the Opam dev test for this branch and master are still failing: in examples/stlc/verif.v, there is a use of Arith.Max.le_max_l and Arith.Max.le_max_r. Arith.Max was deprecated in coq 8.16.0 and appears to have been removed in coq dev. 

The intended replacement, PeanoNat.Nat.le_max_l and PeanoNat.Nat.le_max_r, I believe are in all versions from 8.9 onwards. I am going to perform that replacement in this request.

File: plugin/tactic_quickchick.mlg.cppo [hunk 1]
```diff
@@ -224,16 +271,17 @@ let quickchick_goal =
...
  *)
 
+*)
 }
-
 TACTIC EXTEND quickchick
   | ["quickchick"] -> { quickchick_goal }
 END
-
+(*
```
Comment (lemonidas) [inline plugin/tactic_quickchick.mlg.cppo:279]: These should not be commented out.

File: plugin/tactic_quickchick.mlg.cppo [hunk 2]
```diff
@@ -6,10 +6,10 @@ MAKE SURE TO EDIT THE .cppo SOURCE OF THIS FILE RATHER THAN THE GENERATED RESULT
 
 *)
 
-  open Stdarg
```
Comment (lemonidas) [inline plugin/tactic_quickchick.mlg.cppo:9]: Re-enable this, once you re-enable the merge-related tactics.

File: plugin/tactic_quickchick.mli
Comment (lemonidas) [inline plugin/tactic_quickchick.mli:1]: Might be unnecessary once you re-enable the merge-related tactics.

**Review by lemonidas — CHANGES_REQUESTED** [review]
This looks good - once the merge-related tactics (which seem otherwise untouched) are re-enabled, I'll approve.

**Review by lemonidas — APPROVED** [review]

**Pull request merged**